### PR TITLE
SDL draw updates

### PR DESCRIPTION
**Pull request by mariotaku**
### Description of the feature or fix

SDL_RenderSetClipRect will slow render down especially on low end devices. This is for a little bit more performance improvement.
Also, it adds polygon drawing logic

### Checkpoints
- [x] Follow the [styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [x] Run `code-format.py` from the `scripts` folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the documentation

### CODE REVIEW

File: src/draw/lv_draw_mask.c [hunk 1]
```diff
@@ -1218,6 +1250,79 @@ LV_ATTRIBUTE_FAST_MEM static lv_draw_mask_res_t lv_draw_mask_map(lv_opa_t * mask
     return LV_DRAW_MASK_RES_CHANGED;
 }
 
+LV_ATTRIBUTE_FAST_MEM static lv_draw_mask_res_t lv_draw_mask_polygon(lv_opa_t * mask_buf, lv_coord_t abs_x,
```
Comment (kisvegabor) [inline src/draw/lv_draw_mask.c:1253]: Does it support concave polygons like this?
image

**Review by kisvegabor — COMMENTED** [review]
Thanks!

File: src/draw/lv_draw_mask.c (hunk 1 above)
Comment (mariotaku) [reply src/draw/lv_draw_mask.c:1253]: It supports convex polygon only. Should LVGL support concave polygons?

Comment (kisvegabor) [reply src/draw/lv_draw_mask.c:1253]: It should, but doesn't support at this moment :laughing: 

Supporting convex only is fine, as it's in pair with current features.

**Comment by kisvegabor** [comment]
Please fix the code formatting issue, and I'll merge it! :slightly_smiling_face:

**Comment by mariotaku** [comment]
done :)

File: src/draw/lv_draw_mask.c (hunk 1 above)
Comment (mariotaku) [reply src/draw/lv_draw_mask.c:1253]: I'll support this in later patches. So users can use mask directly anywhere (without point limit).

**Comment by kisvegabor** [comment]
Merging thanks.

**Pull request merged**